route/route.go

```diff
@@ -330,6 +363,124 @@ func (r *Router) batch(w http.ResponseWriter, req *http.Request) {
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+	if dataset == "" {
+		r.Logger.Error().Logf("Received OTLP request without Honeycomb dataset header")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	apiHost, err := r.Config.GetHoneycombAPI()
+	if err != nil {
+		r.Logger.Error().Logf("Unable to retrieve APIHost from config while processing OTLP batch")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	for _, resourceSpan := range req.ResourceSpans {
+		resourceAttrs := make(map[string]interface{})
+
+		if resourceSpan.Resource != nil {
+			addAttributesToMap(resourceAttrs, resourceSpan.Resource.Attributes)
+		}
+
+		for _, librarySpan := range resourceSpan.InstrumentationLibrarySpans {
+			library := librarySpan.InstrumentationLibrary
+			if library.Name != "" {
+				resourceAttrs["library.name"] = library.Name
+			}
+			if library.Version != "" {
+				resourceAttrs["library.version"] = library.Version
+			}
+
+			for _, span := range librarySpan.GetSpans() {
+				traceID := bytesToTraceID(span.TraceId)
+				spanID := hex.EncodeToString(span.SpanId)
+				timestamp := time.Unix(0, int64(span.StartTimeUnixNano)).UTC()
+
+				eventAttrs := map[string]interface{}{
+					"trace.trace_id": traceID,
+					"trace.span_id":  spanID,
+					"type":           getSpanKind(span.Kind),
+					"name":           span.Name,
+					"duration_ms":    float64(span.EndTimeUnixNano-span.StartTimeUnixNano) / float64(time.Millisecond),
+					"status_code":    int32(span.Status.Code),
+				}
+				if span.ParentSpanId != nil {
+					eventAttrs["trace.parent_id"] = hex.EncodeToString(span.ParentSpanId)
+				}
+				if r.getSpanStatusCode(span.Status) == trace.Status_STATUS_CODE_ERROR {
+					eventAttrs["error"] = true
+				}
+				if len(span.Status.Message) > 0 {
+					eventAttrs["status_message"] = span.Status.Message
+				}
+				if span.Attributes != nil {
+					addAttributesToMap(eventAttrs, span.Attributes)
+				}
+
+				sampleRate := 1
+				if eventAttrs["sampleRate"] != nil {
+					switch eventAttrs["sampleRate"].(type) {
+					case string:
+						sampleRate, err = strconv.Atoi(eventAttrs["sampleRate"].(string))
+						if err != nil {
+							r.Logger.Info().WithField("Unable to parse sampleRate in OTLP span attribute", eventAttrs["sampleRate"])
+						}
+					case int:
+						sampleRate = eventAttrs["sampleRate"].(int)
+					}
+					// remove sampleRate from event fields
+					delete(eventAttrs, "sampleRate")
+				}
+
+				// copy resource attributes to event attributes
+				for k, v := range resourceAttrs {
+					eventAttrs[k] = v
+				}
+
+				event := &types.Event{
+					Context:    ctx,
+					APIHost:    apiHost,
+					APIKey:     apiKey,
+					Dataset:    dataset,
+					SampleRate: uint(sampleRate),
+					Timestamp:  timestamp,
+					Data:       eventAttrs,
+				}
+
+				err := r.processEvent(event, requestId)
+				if err != nil {
+					// log failure? We don't send per-event responses back to sender for OTLP
```

Comment:
Why not log? If we're worried about too many logs we could add a metric instead.

Reply:
Not sure what to do. Log's are useful here as it would give context on _why_ it failed, but potentially could generate a lot of noise.

Reply:
Add log entry in commit [1d4df3e](https://github.com/honeycombio/refinery/pull/202/commits/1d4df3ed5ba2667176b479f2bf7c1ef69f754ab8)